clients/src/main/java/org/apache/kafka/clients/admin/ElectLeadersResult.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.admin;
+
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.annotation.InterfaceStability;
+import org.apache.kafka.common.internals.KafkaFutureImpl;
+
+/**
+ * The result of {@link AdminClient#electLeaders(ElectionType, Collection, ElectLeadersOptions)}
+ *
+ * The API of this class is evolving, see {@link AdminClient} for details.
+ */
+@InterfaceStability.Evolving
+final public class ElectLeadersResult {
```

Comment:
Might be convenient for users to have all the `all()` API?

By the way, do you think it's worth fixing the issues with the old API? We can do this separately, just wondering if you think it's worthwhile.

Reply:
Which issue specifically? That the user need to create `N` futures to get all of the errors?

If so, I think the user should just use `electLeaders` instead.

Reply:
Oh, I was referring to the buggy/inconsistent handling of the partitions=null case.

Reply:
Is it reasonable to add the `all()` API?

Reply:
Sure. We have `partitions: Future[Map[TopicPartition, Optional[Throwable]]]`. Are you suggesting renaming this to `all`? Or do you want to add a new method `all: Future[Unit]` that throws if any of the partitions is set to a `Throwable`?

Reply:
I think it would be nice to have both. Many (maybe most?) of the AdminClient APIs offer `all()` for convenience.